silverback/subscriptions.py

```diff
@@ -0,0 +1,150 @@
+import asyncio
+import json
+from enum import Enum
+from typing import AsyncGenerator, Dict, List, Optional
+
+from ape.logging import logger
+from websockets import client as ws_client
+
+
+class SubscriptionType(Enum):
+    BLOCKS = "newHeads"
+    EVENTS = "logs"
+
+
+class Web3SubscriptionsManager:
+    websocket_reconnect_max_tries: int = 3
+    rpc_response_timeout_count: int = 10
+    subscription_polling_time: float = 0.1  # secs
+
+    def __init__(self, ws_provider_uri: str):
+        # TODO: Temporary until a more permanent solution is added to ProviderAPI
```

Comment:
is the TODO still needed here

Reply:
The goal is eventually all providers can leverage websockets for streaming if they have them, and fall back to polling if they don't 

Then silverback can simplify a lot